### ISSUE TITLE
sportwatch data implementing AAPs

### ISSUE THREAD

**Issue by gazrenzo** [nightscout/AndroidAPS#2817]
I apologize if I did something wrong, but this is all new for me. 

I am an alpinist with T1D who struggle to be on range while doing my up and downhill activities.
I am currently studying to start with an AAPS system and looking if some other datas implementation could fix some recurrent high and lows during my activities. These are obviously not costant in energy expenditure demand, giving that I alternate uphill and downhill with different steepness.

As I read, basing the adjustment on the HR o step count variations could have some trouble in the transitions periods, both from inactivity to activity and from low to high intensity activity (source: Diabetech).

I was thinking that, in my case, correlating altitude variations/time with energy expenditure (coming from a sport watch for example) and to ISF, could be suitable to tell the algoritm what will happen during the future activity.
Of course having the possibility to confront the energy expenditure forecast with the effective one registered by the watch will make the machine to better predict the next one.

I recognize that this approach will still need some previous input by the user, but will maybe show better result in terms of control.
I write these lines to understand if someone already tried out this route or think he or she would be interested in trying to figure out together how to make it possible.

**Comment by schmadde** [comment]
There is at least another discussion here revolving around getting the necessary data into AAPS first. One of the problems is that there is no one "data hub" for that except maybe for google fit and apple health which unfortunately are not "realtime" so not really useable. Another problem is that they concentrate on Heart Rate only, which correlates, but not really well. 

The input parameters and their effect on insuline sensitivity are not well known, at least not quatitatively. But I am also very interested in getting to know this better, since I also like to do endurance sports (cycling, running and hiking). 

I see that there is a big difference depending on how intensive the workout is (low intensity: not much effect, medium intensity: insuline sensitivity increases, high intensity: sensitivity **de**creases (BG Rise) but after the workout it increases (BG falls)), how long the workout is (the longer, the higher the sensitivity and also the longer the "muscle refill effect"). But I could not find any waterproof pattern to all that. If there is any scientific analysis on that I would like to know. 

In short: I am interested in discussing/analysing this, but do not know a good way to handle it yet.
